indico/modules/rb_new/client/js/components/TimeRangePicker.jsx

```diff
@@ -0,0 +1,198 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import moment from 'moment';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Dropdown} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+import {toMoment, serializeTime} from '../util';
+
+import './TimeRangePicker.module.scss';
+
+
+function _humanizeDuration(duration) {
+    const hours = duration.hours();
+    const minutes = duration.minutes();
+    if (hours === 1 && minutes === 0) {
+        return Translate.string('1 hour');
+    } else if (hours !== 0) {
+        return Translate.string('{time} hours', {time: hours + (minutes / 60)});
+    } else {
+        return Translate.string('{time} minutes', {time: minutes});
+    }
+}
+
+export default class TimeRangePicker extends React.Component {
+    static propTypes = {
+        startTime: PropTypes.object,
+        endTime: PropTypes.object,
+        onChange: PropTypes.func.isRequired,
+    };
+
+    static defaultProps = {
+        startTime: moment().startOf('hour').add(1, 'h'),
+        endTime: moment().startOf('hour').add(2, 'h')
+    };
+
+    constructor(props) {
+        super(props);
+
+        const {startTime, endTime} = this.props;
+        const startOptions = this.generateStartTimeOptions();
+        const endOptions = this.generateEndTimeOptions(startTime);
+        const duration = moment.duration(endTime.diff(startTime));
+        this.addOptionIfMissing(startOptions, serializeTime(moment(startTime)));
+        this.addOptionIfMissing(endOptions, serializeTime(moment(endTime)));
+        this.state = {
+            startTime,
+            endTime,
+            startOptions,
+            endOptions,
+            duration
+        };
+    }
+
+    generateStartTimeOptions = () => {
+        const options = [];
+        const end = moment().endOf('day');
+        const next = moment().startOf('day');
+        let serializedNext;
+        while (next < end) {
+            serializedNext = serializeTime(moment(next));
+            options.push({key: serializedNext, value: serializedNext, text: serializedNext});
+            next.add(30, 'm');
+        }
+        return options;
+    };
+
+    generateEndTimeOptions = (start) => {
+        const options = [];
+        const end = moment().endOf('day');
+        const next = moment(start).add(30, 'm');
+        let serializedNext, duration, text;
+        while (next < end) {
+            duration = _humanizeDuration(moment.duration(next.diff(start)));
+            serializedNext = serializeTime(moment(next));
+            text = <div>{serializedNext} <div styleName="duration">({duration})</div></div>;
```

Comment:
`const text = ...`; no need to define it outside, `const` and `let` are block-scoped.